config/params/network_config_develop.go

```diff
@@ -0,0 +1,38 @@
+//go:build develop
+
+package params
+
+import (
+	"sync"
+
+	"github.com/mohae/deepcopy"
+)
+
+var networkConfigLock sync.Mutex
```

Comment:
@jsvisa I'm confused about the 2 different impls of `NetworkConfig` in `config/params/network_config_develop.go` vs `config/params/network_config.go`
and the corresponding differences in `SetupTestConfigCleanup`.

It seems like the main difference is having a lock that is only used in the _develop versions. Why do we need that lock in one build target and not the other?

Reply:
In develop/test builds: The networkConfig can be modified at runtime via OverrideBeaconNetworkConfig(). Since multiple goroutines might concurrently:

Read the config via BeaconNetworkConfig()
Override the config via OverrideBeaconNetworkConfig()
Copy the config via Copy() method
While in production builds: The networkConfig is effectively read-only after initialization, so no needs to take care of the datarace.